static/js/user_groups.ts

```diff
@@ -176,3 +176,26 @@ export function get_recursive_subgroups(user_group_id: number): Set<number> | un
     }
     return subgroup_ids;
 }
+
+export function is_user_in_group(user_group_id: number, user_id: number): boolean {
+    const user_group = user_group_by_id_dict.get(user_group_id);
+    if (user_group === undefined) {
+        blueslip.error(`Could not find user group with ID ${user_group_id}`);
+        return false;
+    }
+    if (is_direct_member_of(user_group_id, user_id)) {
+        return true;
+    }
+
+    const subgroup_ids = get_recursive_subgroups(user_group_id);
+    if (subgroup_ids === undefined) {
```

Comment:
Under what circumstance do we expect it to return `undefined`? I would sorta hope that once we've successfully fetched the `user_group` object above, that it's not possible to fail in that way. If it's just the possibility that the user group ID itself was invalid, which we've already checked above, one idea would be to have `get_recursive_subgroups` accept a group rather than and ID as its parameter. (We can add a `get_recursive_subgroups_by_id` if we need to accept an ID in some future code path).

Reply:
`get_recursive_subgroups` also returns `undefined` if we cannot get group object of any one of the subgroup id (we need the object to get further subgroups of that group).

I also thought of passing directly group instead of id to `get_recursive_subgroups` but thought we might need to accept ID in future. But adding separate function to accept ID in future if required seems nice idea.

Reply:
Cool. I think this is probably fine to do as a follow-up if we can make it work. 

We will need to consider the possibility that in some situations, we will not have access to subgroups of a group, if we decide to add support for limiting access to viewing a group exists or its membership. Not something to worry about for now, I think, since it's possible we won't go for that type of feature.